src/sas/sasgui/guiframe/gui_manager.py

```diff
@@ -2573,20 +2570,26 @@ def save_data2d(self, data, fname):
             ext_num = dlg.GetFilterIndex()
             if ext_num == 0:
                 ext_format = '.dat'
+            elif ext_num == 1:
+                ext_format = '.h5'
             else:
                 ext_format = ''
             path = os.path.splitext(path)[0] + ext_format
             mypath = os.path.basename(path)
 
             # Instantiate a loader
             loader = Loader()
-
-            ext_format = ".dat"
-            if os.path.splitext(mypath)[1].lower() == ext_format:
+            if os.path.splitext(mypath)[1].lower() == '.dat':
```

Comment:
Since you are using it twice...

    ext = os.path.splitext(mypath)[1].lower()
    if ext == '.dat':
        ...
    elif ext == '.h5':
        ...